### DISCUSSION TITLE
Query on summary-rollup-syncts Druid data source

### DISCUSSION THREAD

**Discussion by spsac08** [Sunbird-Obsrv/Community#19] [Query]
Hi Team,

We are trying to create reports for Learn related data on the top of  **summary-rollup-syncts**  Druid data source,
And we have a supervisor running for this data source in Druid,
We wanted to know how do we enable this data source in druid and what type of data to be pushed.

[@sowmya-dixit](https://github.com/sowmya-dixit)  Can you please help us on this ?

cc [@aks30](https://github.com/aks30), [@kiranharidas187](https://github.com/kiranharidas187), [@vijiurs](https://github.com/vijiurs),[@Shakthieshwari](https://github.com/Shakthieshwari)

**Comment by spsac08** [comment]
Hi Team,
I need help on this query ( Understading ), can anyone please assist me ?

**Reply by rhwarrier** [reply]
@sowmya-dixit  @anandp504 Could you please take a look ?

**Reply by anandp504** [reply]
@spsac08 Druid Supervisors pull data from a Kafka topic which is configured in the ingestion spec. You will need to push your data into the Kafka topic for it to be ingested and queried. Please provide more information on your query regarding enabling the data source.

**Reply by spsac08** [reply]
@anandp504 We have a supervisor running in Druid and we added injection spec to it.
Our query is just what type of data we need to push to Kafka so that we can create this data source,
 if any data source is dependent on this.

**Reply by sowmya-dixit** [reply]
@spsac08 These **summary-rollup-syncts** data are derived summary events generated out of raw telemetry events which are generated by user interactions from UI. Complete data pipeline and sunbird data products set up needs to be available to generate these kind of summary data. Please provide more details on your requirement of enabling this data source.

**Reply by spsac08** [reply]
@sowmya-dixit Thank you, and can you please provide us which other data pipeline needed and sunbird data product is required?
can you please give us some git repo links to generate the summary rollup syncts datasource.

**Reply by sowmya-dixit** [reply]
@spsac08  Please share us more details on your requirement like - are you trying to set this up in your own separate env? If yes does it already have telemetry events getting generated? What other set ups are already available related to Obsrv? 
Please find more details in below docs which also has git repo details,
Data pipeline - https://obsrv.sunbird.org/learn/product-and-developer-guide/data-pipeline
Summary Data Product - https://obsrv.sunbird.org/learn/product-and-developer-guide/summarisers

**Reply by spsac08** [reply]
@sowmya-dixit Yes, We are trying to set it up in own separate env, and we are also using telemetry services, and data getting generated and Flink jobs and Druid Clusters with ingestion Tasks

**Comment by spsac08** [comment]
Hi @sowmya-dixit,
All our environment stepup like Flink Job, Data Products and Analytics stepup is done and completed.
We need to know what type of data to submit, as in course etc. And where it will be stored in places like Kafka, mongo db, Casandra etc..

**Reply by spsac08** [reply]
@sowmya-dixit can you please help us out here?

**Reply by spsac08** [reply]
Hi, @sowmya-dixit Good morning,
we are blocked in this, could you please help us here?

**Reply by sowmya-dixit** [reply]
Hi @spsac08 ,
The telemetry data generated and processed via Flink jobs will be backed up to blob store using Secor. Is secor backups set up also done?
If yes, We have a generic data product called WorkFlow Summariser(https://obsrv.sunbird.org/learn/product-and-developer-guide/summarisers), which generates summary data which is pushed to kafka, processed via the data pipeline Flink jobs and then ingested to Druid via supervisor which you have already running.

Please make sure below setups are also done to enable summary datasource -
- Telemetry data backup to blob store using Secor
- Schedule **wfs** (Workflow Summariser - Data Product) in Spark server to run daily

**Reply by spsac08** [reply]
Thank you @sowmya-dixit, 
will get back to you if we need any help.

**Reply by spsac08** [reply]
HI @sowmya-dixit 

1. Secor backups setup is done and running. 
2. how do we check if **wfs** (Workflow Summarizer - Data Product) is running or not?

**Reply by spsac08** [reply]
@sowmya-dixit can you please help us with this, How do we check the **Workflow Summarizer - Data Product**  is running or not?

**Reply by sowmya-dixit** [reply]
@spsac08 
Please check if crontab is setup for WFS, it will be run using this script in spark server - `/mount/data/analytics/scripts/run-job.sh wfs`
If not please setup this script to run everyday. It can also be run manually once to check if it runs fine and execution logs can be found under - `/mount/data/analytics/logs/data-products/<CurrentDate>-job-execution.log` file.

**Reply by spsac08** [reply]
Thank you @sowmya-dixit

**Reply by spsac08** [reply]
we can see data in druid, **Thank you very much** @sowmya-dixit, the last request is if there is any document to understand the data can you please point out where will get it?

**Reply by sowmya-dixit** [reply]
@spsac08 - You can find more details on the summary data in this document - [https://github.com/project-sunbird/sunbird-core-dataproducts/wiki/%5BData-Product%5D-Workflow-Summarizer#output](https://github.com/project-sunbird/sunbird-core-dataproducts/wiki/%5BData-Product%5D-Workflow-Summarizer#output)
